core-rust/jni-export/src/constants.rs

```diff
@@ -0,0 +1,164 @@
+};
+use node_common::config::{
+    DEFAULT_MEMPOOL_MAX_TOTAL_TRANSACTIONS_SIZE, DEFAULT_MEMPOOL_MAX_TRANSACTION_COUNT,
+};
+use node_common::java::jni_sbor_coded_call;
+use radix_engine_constants::{DEFAULT_COST_UNIT_LIMIT, DEFAULT_MAX_TRANSACTION_SIZE};
+use state_manager::priority_mempool::MEMPOOL_TRANSACTION_OVERHEAD_FACTOR_PERCENT;
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultMaxVertexTransactionCount(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| {
+        DEFAULT_MAX_VERTEX_TRANSACTION_COUNT
+    })
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultMaxTotalVertexTransactionsSize(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| {
+        DEFAULT_MAX_TOTAL_VERTEX_TRANSACTIONS_SIZE
+    })
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultMaxTotalVertexExecutionCostUnitsConsumed(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| {
+        DEFAULT_MAX_TOTAL_VERTEX_EXECUTION_COST_UNITS_CONSUMED
+    })
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultMempoolMaxTotalTransactionsSize(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| {
+        DEFAULT_MEMPOOL_MAX_TOTAL_TRANSACTIONS_SIZE
+    })
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultMempoolMaxTransactionCount(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| {
+        DEFAULT_MEMPOOL_MAX_TRANSACTION_COUNT
+    })
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultMaxTransactionSize(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| DEFAULT_MAX_TRANSACTION_SIZE)
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getDefaultCostUnitLimit(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| DEFAULT_COST_UNIT_LIMIT)
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_environment_NodeConstants_getMempoolTransactionOverheadFactorPercent(
+    env: JNIEnv,
+    _class: JClass,
+    request_payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, request_payload, |_: ()| {
+        MEMPOOL_TRANSACTION_OVERHEAD_FACTOR_PERCENT
+    })
+}
```

Comment:
don't you prefer a single native method returning `struct AllTheseConsts{...}`?

(I know it will need an ugly `static {...}` initializer on java side)

Reply:
I thought about that, but decided to go for the current pattern of `ScryptoConstants`. What do you think of this refactor in a future more general `jni` refactor?

Reply:
If this is your plan in a refactoring that is really "coming soon", then I can live with the current shape for a moment 👍